### PR TITLE
Support App Scan Enterprise

### PR DESCRIPTION
**Pull request by geoffwhittington**
Support App Scan Enterprise
Add tests for task_status_mapping

### CODE REVIEW

File: sdetools/analysis_integration/tests/base_integration_test.py [hunk 1]
```diff
@@ -115,6 +115,42 @@ def test_valid_import_behaviour(self):
...
+            'fail': ['BAD', 'DONE', 'NA'],
+            'pass': ['BAD','NA'],
+        }
+        for verification, statuses in invalid_mappings.iteritems():
+            for status in statuses:
+                self.init_integrator()
+                self.config['task_status_mapping'] = {verification: status}
+                try:
+                    self.init_data()
+                    self.assertTrue(False, 'Invalid task_status_mapping (%s=>%s) not detected' % (verification, status))
```
Comment (funkaoshi) [inline sdetools/analysis_integration/tests/base_integration_test.py]: I believe unittest2 has a method called self.fail() you can use here.

File: sdetools/analysis_integration/tests/base_integration_test.py [hunk 2]
```diff
@@ -115,6 +115,42 @@ def test_valid_import_behaviour(self):
...
+                except UsageError:
+                    self.assertTrue(True)
```
Comment (funkaoshi) [inline sdetools/analysis_integration/tests/base_integration_test.py]: You can just `pass`. This is a no-op anyway.

File: sdetools/analysis_integration/tests/base_integration_test.py [hunk 3]
```diff
@@ -115,6 +115,42 @@ def test_valid_import_behaviour(self):
...
+        }
+        for verification, statuses in valid_mappings.iteritems():
+            for status in statuses:
+                self.init_integrator()
+                self.config['task_status_mapping'] = {verification: status}
+                try:
+                    self.init_data()
+                    self.assertTrue(True)
+                except UsageError:
+                    self.assertTrue(False, 'Valid task_mapping (%s => %s) improperly disallowed' %
```
Comment (funkaoshi) [inline sdetools/analysis_integration/tests/base_integration_test.py]: Same, can do self.fail() here, I think.

File: sdetools/modules/import_appscan/appscan_enterprise_xml_importer.py [hunk 4]
```diff
@@ -0,0 +1,63 @@
+from sdetools.analysis_integration.base_integrator import BaseXMLImporter, BaseContentHandler
+
+
+class AppScanEnterpriseXMLContent(BaseContentHandler):
+    def __init__(self):
+        self.saw_app_scan_node = False
+        self.in_report_node = False # top-level node
```
Comment (funkaoshi) [inline sdetools/modules/import_appscan/appscan_enterprise_xml_importer.py]: Can only one of these flags be true at a time? Might want to make these four variables a single enumerated type instead.

File: sdetools/modules/import_appscan/tests/appscan_test.py [hunk 5]
```diff
@@ -16,3 +17,11 @@ def setUpClass(cls):
 
     def expected_number_of_findings(self):
         return 2
+
+    def test_invalid_edition(self):
+        self.config['edition'] = 'unknown'
+        try:
+            self.init_data()
+            self.assertTrue(False, 'Unsupported AppScan edition: %s' % self.config['edition'])
```
Comment (funkaoshi) [inline sdetools/modules/import_appscan/tests/appscan_test.py]: `self.fail()` here as well.

File: sdetools/modules/import_appscan/tests/appscan_test.py [hunk 6]
```diff
@@ -16,3 +17,11 @@ def setUpClass(cls):
...
+        except UsageError:
+            self.assertTrue(True)
```
Comment (funkaoshi) [inline sdetools/modules/import_appscan/tests/appscan_test.py]: Replace this with `pass`.

File: sdetools/sdelib/testlib/sde_response_generator.py [hunk 7]
```diff
@@ -77,6 +79,20 @@ def get_phases(self, target, flag, data, method):
         else:
             self.raise_error('401')
 
+    def get_taskstatuses(self, target, flag, data, method):
+        if not flag:
```
Comment (funkaoshi) [inline sdetools/sdelib/testlib/sde_response_generator.py]: If flag isn't set we always raise a 401? Might want to give that variable a better name?

File: sdetools/analysis_integration/tests/base_integration_test.py (hunk 1 above)
Comment (geoffwhittington) [inline sdetools/analysis_integration/tests/base_integration_test.py]: We're using unittest now

File: sdetools/modules/import_appscan/appscan_enterprise_xml_importer.py (hunk 4 above)
Comment (geoffwhittington) [inline sdetools/modules/import_appscan/appscan_enterprise_xml_importer.py]: These boolean flags track where we are in the XML - all or 1 could be true at any time.

File: sdetools/sdelib/testlib/sde_response_generator.py (hunk 7 above)
Comment (geoffwhittington) [inline sdetools/sdelib/testlib/sde_response_generator.py]: My preference is to resolve this legacy issue in another PR

**Comment by funkaoshi** [comment]
Yeah this seems fine. I think there might be a tidier way of loading / picking the importer, but I wouldn't block merging on that.

**Comment by geoffwhittington** [comment]
We have some ideas cooking in the kitchen but its beyond the scope of this PR, unfortunately. It works and its not going to be a big deal in the grand scheme of things

**Pull request merged**